sammba/registration/struct.py

```diff
@@ -426,75 +456,32 @@ def anats_to_common(anat_filenames, write_dir, brain_volume,
-    # The input transform is the former warp and needs to be concatenated to
-    # IDENT initially; I forget why, I think it is to avoid some weird bug.
-    if len(nonlinear_levels) > 1:
-        previous_warp_files = warp_files
-        warped_files = []
-        warp_files = []
-        for warp_file, centered_head_file in zip(previous_warp_files,
-                                                 centered_head_files):
-            out_nwarp_cat = nwarp_cat(
-                in_files=[('IDENT', out_tstat_warp_head.outputs.out_file),
-                          warp_file], out_file='iniwarp.nii.gz')
-            out_qwarp = qwarp(
-                in_file=centered_head_file,
-                base_file=out_tstat_warp_head.outputs.out_file,
-                nmi=True,
-                noneg=True,
-                iwarp=True,
-                weight=out_mask_tool.outputs.out_file,
-                iniwarp=[out_nwarp_cat.outputs.out_file],
-                inilev=nonlinear_levels[0],
-                maxlev=nonlinear_levels[1],
-                out_file=fname_presuffix(centered_head_file,
-                                         suffix='_warped2'))
-            warp_files.append(out_qwarp.outputs.source_warp)
-            warped_files.append(out_qwarp.outputs.warped_source)
-
-        out_tcat = tcat(in_files=warped_files,
-                        out_file='warped_2iters_heads.nii.gz')
-        out_tstat_warp_head = tstat(in_file=out_tcat.outputs.out_file,
-                                    outputtype='NIFTI_GZ')
-
     ###########################################################################
     # Using previous files and concatenated transforms can be exploited to
     # avoid building up reslice errors.
-    # Warp with mini-patch
-    # In this particular case, minpatch=75 corresponds to a level of 4
-    if len(nonlinear_levels) > 2:
-        if nonlinear_minimal_patch is None:
-            nonlinear_minimal_patch = 75
+    # Warp to maxlev
+    if len(nonlinear_levels) > 1:
 
-        for n_iter, inilev in enumerate(nonlinear_levels[2:]):
+        for n_iter, maxlev in enumerate(nonlinear_levels[1:]):
+            inilev = nonlinear_levels[n_iter] + 1
             previous_warp_files = warp_files
             warped_files = []
             warp_files = []
             for warp_file, centered_head_file in zip(previous_warp_files,
                                                      centered_head_files):
-                suffixed_file = fname_presuffix(
+                out_file = fname_presuffix(
                     centered_head_file,
-                    suffix='_warped{}'.format(n_iter + 3))
-                if n_iter == len(nonlinear_levels):
-                    out_file = os.path.join(write_dir,
-                                            os.path.basename(suffixed_file))
-                else:
-                    out_file = suffixed_file
+                    suffix='_warped{}'.format(n_iter + 1 + len(nonlinear_levels)))
```

Comment:
you mean out_file first iteration will be suffixed _warped3 if you entered 2 non linear levels ? I am not following ...

Reply:
this is consistent with what was already there, naming files based on the iteration of the loop rather than the user-chosen -maxlev or -minpatch